docs/references/go/verifying-sessions.mdx

```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
+Their difference is that the `RequireHeaderAuthorization` calls `WithHeaderAuthorization` under the hood, but responds
+with HTTP 403 Forbidden if it fails to detect valid session claims.
+
+Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
+
+<InjectKeys>
+```go
 import (
-	"net/http"
-	"strings"
+  "fmt"
+  "net/http"
 
-	"github.com/clerk/clerk-sdk-go/clerk"
+  "github.com/clerk/clerk-sdk-go/v2"
+  clerkhttp "github.com/clerk/clerk-sdk-go/v2/http"
 )
 
 func main() {
-	client, _ := clerk.NewClient(`{{secret}}`)
-
-	http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
-		// get session token from Authorization header
-		sessionToken := r.Header.Get("Authorization")
-		sessionToken = strings.TrimPrefix(sessionToken, "Bearer ")
-
-		// verify the session
-		sessClaims, err := client.VerifyToken(sessionToken)
-		if err != nil {
-			w.WriteHeader(http.StatusUnauthorized)
-			w.Write([]byte("Unauthorized"))
-			return
-		}
-
-		// get the user, and say welcome!
-		user, err := client.Users().Read(sessClaims.Claims.Subject)
-		if err != nil {
-			panic(err)
-		}
-
-		w.Write([]byte("Welcome " + *user.FirstName))
-	})
-
-	http.ListenAndServe(":8080", nil)
+  clerk.SetKey(`{{secret}}`)
+
+  mux := http.NewServeMux()
+  mux.HandleFunc("/", publicRoute)
+  protectedHandler := http.HandlerFunc(protectedRoute)
+  mux.Handle(
+    "/protected",
+    clerkhttp.WithHeaderAuthorization()(protectedHandler),
+  )
+
+  http.ListenAndServe(":3000", mux)
+}
+
+func publicRoute(w http.ResponseWriter, r *http.Request) {
+  w.Write([]byte(`{"access": "public"}`))
 }
-```
 
+func protectedRoute(w http.ResponseWriter, r *http.Request) {
+  claims, ok := clerk.SessionClaimsFromContext(r.Context())
+  if !ok {
+    w.WriteHeader(http.StatusUnauthorized)
+    w.Write([]byte(`{"access": "unauthorized"}`))
+    return
+  }
+  fmt.Fprintf(w, `{"user_id": "%s"}`, claims.Subject)
+}
+```
 </InjectKeys>
 
-## Using middleware
+## Custom token verification
 
-The Clerk SDK also provides a simple middleware that adds the active session to the request's context.
+It is recommended to use the middleware functions for verifying Clerk sessions in an HTTP context.
 
-<InjectKeys>
+The middleware guarantees better performance and efficiency by making the minimum necessary requests to the Clerk Backend API.
 
-```go
-package main
+Verifying a Clerk session requires providing a JSON Web Key. The middleware fetches the key once and caches it.
+
+If you decide to verify the session token on your own, please be mindful of API rate limits. It is recommended to cache your JSON Web Key and invalidate the cache only when the key gets rolled.
```

Comment:
```suggestion
If you decide to verify the session token on your own, please be mindful of API rate limits. It is recommended to cache your JSON Web Key and invalidate the cache only when a replacement API key is generated.
```

Just wanted to make this slightly less technical, as I'd never heard of the term "Rolling" an API key until I looked it up just now. If you think that's a common enough term, you can ignore this suggestion, as I'll defer to your expertise. Just wanna make sure this content is accessible to all readers 👍

Reply:
Good thinking. Updated.